### PR TITLE
feat: setup library environment #1

### LINKED ISSUE #1: Setup build and publish environment for the library
- https://docs.gradle.org/current/samples/sample_building_kotlin_libraries.html
- https://docs.gradle.org/current/userguide/publishing_setup.html

### PR DESCRIPTION
**Pull request by HoucemKacem**
Build scan: https://scans.gradle.com/s/7naadonrlg5ty
Still need to set maven central repository credentials.
@bobeal do we already have the maven repo ?

### CODE REVIEW

**Comment by bobeal** [comment]
> Still need to set maven central repository credentials.
> @bobeal do we already have the maven repo ?

For now, we'll publish it as a GitHub package.

See for instance what is done in https://github.com/bobeal/ksenml (config in build.gradle.kts and publish.yml Github action to publish the package).

File: lib/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+plugins {
+    // Apply the org.jetbrains.kotlin.jvm Plugin to add support for Kotlin.
+    id("org.jetbrains.kotlin.jvm") version "1.4.20"
```
Comment (bobeal) [inline lib/build.gradle.kts]: why not using the last Kotlin version (1.4.31)?

File: lib/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    implementation("com.google.guava:guava:29.0-jre")
+
+    // Use the Kotlin test library.
+    testImplementation("org.jetbrains.kotlin:kotlin-test")
+
+    // Use the Kotlin JUnit integration.
+    testImplementation("org.jetbrains.kotlin:kotlin-test-junit")
+
+    // This dependency is exported to consumers, that is to say found on their compile classpath.
+    api("org.apache.commons:commons-math3:3.6.1")
```
Comment (bobeal) [inline lib/build.gradle.kts]: ?

File: lib/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+dependencies {
+    // Align versions of all Kotlin components
+    implementation(platform("org.jetbrains.kotlin:kotlin-bom"))
+
+    // Use the Kotlin JDK 8 standard library.
+    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
+
+    // This dependency is used internally, and not exposed to consumers on their own compile classpath.
+    implementation("com.google.guava:guava:29.0-jre")
```
Comment (bobeal) [inline lib/build.gradle.kts]: we have this?

Comment (HoucemKacem) [reply lib/build.gradle.kts]: I'll remove these two dependencies, they were generated among the project dependencies, i preferred to keep them for an initial clean setup.

Comment (bobeal) [reply lib/build.gradle.kts]: If you keep not needed things, it's not clean ;)

**Comment by HoucemKacem** [comment]
> For now, we'll publish it as a GitHub package.

I added the github action for publishing as github package.

**Comment by bobeal** [comment]
LGTM

**Pull request merged**